docs/migration/migratingfrommlapi.md

```diff
@@ -0,0 +1,136 @@
+---
+id: migratingfrommlapi
+title: Updating to the Unity Package
+sidebar_label: Updating to the Unity Package
+description: Learn how to upgrade your current MLAPI installation to the Unity MLAPI package.
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+:::important
+This upgrade guide targets project using the MLAPI version which was installed by the MLAPI installer or by cloning the source code from Github.
+If your project uses a Package Manager version of MLAPI then this guide is not relevant.
+:::
+
+The aqcuisiton of MLAPI has been a unique event in Unity history. In our efforts to integrate MLAPI into the Unity ecosystem, we are providing it in a Unity package.
+
+What this means for you is:
+- MLAPI will be easily accessible from the Unity Editor in the Package Manager window.
+- Updating MLAPI in the future will work directly from the package manager.
+- You will have full access to the MLAPI source in your project making it easier to debug and inspect MLAPI.
+- Upgrading from the last MLAPI version to the new package version is a bit complicated. We are sorry for that.
+
+## Backup your project
+
+:::important Required
+This step is **required**: Upgrading to the package version of MLAPI can cause issues with your current project. The new version modifies files and locations that will be drastically different than previous MLAPI versions.
+:::
+
+Backup your project using the following recommended methods:
+
+* Create a copy of your entire project folder.
+* Use source control software, like Git. 
+
+:::bestpractice
+We recommend using both methods to backup your project. This gives you a copied project and tracking through committed changes and history.
+:::
+
+##  Upgrade to Unity MLAPI
+
+Manually upgrading from the `.dll` version installed by MLAPI to the new package version breaks all MLAPI component references in your scenes and prefabs. This is the case because MLAPI references components using GUIDs, which work differently for packages than for `.dll`s. 
+
+To smoothly upgrade you to the new version of MLAPI, we have created an upgrade tool.
+
+To start upgrading, add the upgrade tool to your project by using the **Add package from git URL..** option in the Package Manager window. Use the following URL: 
+
+`https://github.com/Unity-Technologies/mlapi-community-contributions.git?path=/com.unity.multiplayer.mlapi-patcher`
+
+
+After installing the patcher package you are good to go. Follow the following steps to upgrade.
+
+## 1.**Install the MLAPI Package**
+Follow the [installation guide](TODO how can I easily link to something telling the user how to install MLApI?) for installing the package version of MLAPI.
+
+After installing the package, you will have error messages in the console, which is expected behavior. MLAPI uses and includes numerous new types, about double of your existing project's types.
```

Comment:
```suggestion
After installing the package, you will have error messages in the console, which is expected behavior because your project now contains two different versions of MLAPI at the same type. No worries, we will later clean up the old MLAPI version.
```
This correction changed the meaning of what I was trying to say. There are duplicate types not many more types. But the initial text wasn't great how about this